### PR TITLE
netty: Removed 4096 min buffer size

### PR DESCRIPTION
**Pull request by AgraVator**
Fixes https://github.com/grpc/grpc-java/issues/11719
MessageFramer.flush() is being called between every message, so messages are never combined and the larger allocation just wastes memory. The change removes the restriction on the min buffer size and hence, prevents the said wastage.

### CODE REVIEW

File: netty/src/test/java/io/grpc/netty/NettyWritableBufferAllocatorTest.java [hunk 1]
```diff
@@ -44,7 +44,6 @@ protected WritableBufferAllocator allocator() {
   public void testCapacityHasMinimum() {
     WritableBuffer buffer = allocator().allocate(100);
     assertEquals(0, buffer.readableBytes());
-    assertEquals(4096, buffer.writableBytes());
```
Comment (ejona86) [inline netty/src/test/java/io/grpc/netty/NettyWritableBufferAllocatorTest.java:47]: Assert that it is 100?

**Review by ejona86 — COMMENTED** [review]
We need to include more context. At the very least this (the commit and the PR description) should have a "Fixes #11719" to associate it with the issue (and it will also auto-close the issue when merging). But we should also summarize _why_ we are making the change. I'd say something like, "MessageFramer.flush() is being called between every message, so messages are never combined and the larger allocation just wastes memory."

File: netty/src/test/java/io/grpc/netty/NettyWritableBufferAllocatorTest.java [hunk 2]
```diff
@@ -44,7 +44,6 @@ protected WritableBufferAllocator allocator() {
...
   }
 
   @Test
```
Comment (shivaspeaks) [inline netty/src/test/java/io/grpc/netty/NettyWritableBufferAllocatorTest.java:49]: We might also want to club this test method with above as there is no minimum buffer size now? It's only the allocated buffer size. We should also rename the new test method.

File: netty/src/test/java/io/grpc/netty/NettyWritableBufferAllocatorTest.java (hunk 1 above)
Comment (AgraVator) [reply netty/src/test/java/io/grpc/netty/NettyWritableBufferAllocatorTest.java:47]: `  @Test
  public void testCapacityIsExactAboveMinimum() {
    WritableBuffer buffer = allocator().allocate(9000);
    assertEquals(0, buffer.readableBytes());
    assertEquals(9000, buffer.writableBytes());
  }`
  Since we already have this, should we remove the testCapacityHasMinimum() test ?

Comment (ejona86) [reply netty/src/test/java/io/grpc/netty/NettyWritableBufferAllocatorTest.java:47]: That's fine.

**Review by ejona86 — APPROVED** [review]
You'll want to "squash and merge" using the github button. It will give you the opportunity to write the commit message, and you'll want to clean it up, because it gives a very poor default (it is a _useful_ default, as it gives you all the words you may want to adjust, but it will need editing).

**Comment by ejona86** [comment]
I see you created the branch in the grpc/grpc-java repo. You should generally have the branch in your own fork/repo instead.
